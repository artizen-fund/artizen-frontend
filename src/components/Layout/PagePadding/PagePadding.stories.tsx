import PagePadding from './'

export default {
  title: 'layout/PagePadding',
  component: PagePadding,
  argTypes: {},
}

export const PagePaddingComponent = (props: any) => {
  return <PagePadding {...props} />
}
