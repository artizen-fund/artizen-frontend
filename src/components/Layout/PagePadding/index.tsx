import styled from 'styled-components'
import { breakpoint, palette } from '@theme'
import { rgba } from '@lib'

interface PagePaddingProps {
  black?: boolean
}

const PagePadding = styled.div<PagePaddingProps>`
  padding: 40px 0;
  ${props =>
    props.black &&
    `
    background: ${rgba(palette.night)};
    @media (prefers-color-scheme: dark) {
      background: ${rgba(palette.slate)};
    }
  `}

  @media only screen and (min-width: ${breakpoint.tablet}px) {
    padding: 48px 0;
  }

  @media only screen and (min-width: ${breakpoint.laptop}px) {
    padding: 64px 0;
  }

  @media only screen and (min-width: ${breakpoint.desktop}px) {
    padding: 80px 0;
  }

  > * {
    max-width: calc(100vw - 48px);
    margin: auto;

    @media only screen and (min-width: ${breakpoint.tablet}px) {
      max-width: 688px;
    }

    @media only screen and (min-width: ${breakpoint.laptop}px) {
      max-width: calc(100vw - 320px);
    }

    @media only screen and (min-width: 1940px) {
      max-width: 1600px;
    }
  }
`

export default PagePadding
