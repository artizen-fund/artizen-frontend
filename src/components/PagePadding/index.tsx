import styled from 'styled-components'
import { breakpoint, palette, Palette } from '@theme'
import { rgba } from '@lib'

interface PagePaddingProps {
  background?: keyof Palette
}

const PagePadding = styled.div<PagePaddingProps>`
  padding: 40px 0;
  ${props =>
    props.background &&
    `
    background: ${rgba(palette[props.background])};
  `}

  @media only screen and (min-width: ${breakpoint.tablet}px) {
    padding: 48px 0;
  }

  @media only screen and (min-width: ${breakpoint.laptop}px) {
    padding: 64px 0;
  }

  @media only screen and (min-width: ${breakpoint.desktop}px) {
    padding: 80px 0;
  }

  > * {
    max-width: calc(100vw - 48px);
    margin: auto;

    @media only screen and (min-width: ${breakpoint.tablet}px) {
      max-width: 688px;
    }

    @media only screen and (min-width: ${breakpoint.laptop}px) {
      max-width: 944px;
    }

    @media only screen and (min-width: ${breakpoint.desktop}px) {
      max-width: 1600px;
    }
  }
`

export default PagePadding
