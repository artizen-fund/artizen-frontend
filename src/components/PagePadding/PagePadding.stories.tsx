import PagePadding from './'

export default {
  title: 'components/PagePadding',
  component: PagePadding,
  argTypes: {},
}

export const PagePaddingComponent = (props: any) => {
  return <PagePadding {...props} />
}
