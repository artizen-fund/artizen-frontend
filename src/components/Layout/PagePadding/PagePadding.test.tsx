import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'
import PagePadding from './'

describe('PagePadding', () => {
  it('renders PagePadding unchanged', () => {
    const { container } = render(<PagePadding>Herp Derp</PagePadding>)
    expect(container).toMatchSnapshot()
  })
})
