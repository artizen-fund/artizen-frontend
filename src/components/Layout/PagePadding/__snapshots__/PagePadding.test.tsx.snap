// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PagePadding renders PagePadding unchanged 1`] = `
<div>
  <div
    class="sc-9199b4ac-0 brREMd"
  >
    Herp Derp
  </div>
</div>
`;
